src/PureScript/Backend/Optimizer/Semantics.purs

```diff
@@ -55,21 +55,17 @@ data BackendSemantics
   | NeutUpdate BackendSemantics (Array (Prop BackendSemantics))
   | NeutLit (Literal BackendSemantics)
   | NeutFail String
-  | NeutBacktrack
   | NeutUncurriedApp BackendSemantics (Array BackendSemantics)
   | NeutUncurriedEffectApp BackendSemantics (Array BackendSemantics)
   | NeutPrimOp (BackendOperator BackendSemantics)
   | NeutPrimEffect (BackendEffect BackendSemantics)
   | NeutPrimUndefined
 
-data SemConditional a = SemConditional a (Maybe (SemTry a) -> a)
-
-type SemTry a = Tuple (Array (Lazy (SemConditional a))) (Maybe (Lazy a))
+data SemConditional a = SemConditional a a
```

Comment:
I think you would want these both to be `Lazy a`, rather than having a `Lazy (SemConditional a)` because evaluating the condition head should not require evaluating the condition body.